utilities-core/src/main/java/uk/gov/justice/services/common/converter/jackson/additionalproperties/AdditionalPropertiesSerializer.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.justice.services.common.converter.jackson.additionalproperties;
+
+import static uk.gov.justice.services.common.converter.jackson.additionalproperties.AdditionalPropertiesHelper.ADDITIONAL_PROPERTIES_NAME;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.impl.ObjectIdWriter;
+import com.fasterxml.jackson.databind.ser.std.BeanSerializerBase;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Map;
+
+
+public class AdditionalPropertiesSerializer extends BeanSerializerBase {
+
+
+    public AdditionalPropertiesSerializer(final BeanSerializerBase source, final String[] toIgnore) {
+        super(source, toIgnore);
+    }
+
+    public BeanSerializerBase withObjectIdWriter(final ObjectIdWriter objectIdWriter) {
+        return null;
+    }
+
+    public BeanSerializerBase withIgnorals(final String[] toIgnore) {
+        return null;
+    }
+
+    public BeanSerializerBase asArraySerializer() {
+        return null;
+    }
+
+    public BeanSerializerBase withFilterId(final Object o) {
+        return null;
+    }
+
+    public void serialize(final Object bean, final JsonGenerator jsonGenerator, final SerializerProvider provider) throws IOException {
+
+        jsonGenerator.writeStartObject();
+        try {
+            final Field apField = bean.getClass().getDeclaredField(ADDITIONAL_PROPERTIES_NAME);
+            apField.setAccessible(true);
```

Comment:
Not sure adding a setter on the bean is a good idea. It breaks encapsulation (I know that that is the point here, but here is a special case). Java 9 changes the access privileges of different modules, but if/when we move to java 9 then we can fix this problem then.

Reply:
No, not a setter. It's likely that there is a getter so possible to do `getAdditionalProperties().put(...)`

`setAccessible(true)` can be a fallback and is better to be accomplished by a warning. (Otherwise it'll be forgotten at the time!)